### ISSUE TITLE
UndocumentedPublicProperty should check the class header for constructor properties

### ISSUE THREAD

**Issue by DanySK** [arturbosch/detekt#2562]
The rule or changes you're looking for might have already been suggested!
Please search in the [issues](https://github.com/arturbosch/detekt/issues) before creating a new one.

## Expected Behavior of the rule

Documentation of public properties may happen in three ways:
1. By mentioning them in the class description
2. By explicitly documenting them with `@property`
3. By adding a Kdoc comment right above them, javadoc style.

It seems to me that Detekt currently forces the third rule.
I propose the behaviour to change in such a way that this class passes the check

```kotlin
/**
 * An example where I mention the property [configuration] in the main text.
 *
 * @property headless is documented with `@property`
 */
data class AlchemistExecutionOptions(
    val configuration: String? = null,
    val headless: Boolean = false,
    /**
     * This kind of documentation is already detekted :)
     */
    val variables: List<String> = emptyList(),
)
```

## Context

With this change, I aim to enforce my developers to find the right balance between documenting the public API and avoid spending too much time and effort into over-documentation just to make Detekt silent.

The issue is clear in a large project such as Alchemist/Alchemist, where many students write Kotlin code. They need to learn how to do it appropriately (because I'll end up using their code), but they also need to be agile, as their main goal is to deliver (and learn).

**Comment by BraisGabin** [comment]
Which version of detekt are you using? I think that this issue was fixed already in 1.7.2

**Comment by schalkms** [comment]
This is exactly how the rule behaves on 1.7.2. Please take a look at the following test cases. What do you find missing @DanySK ?

https://github.com/arturbosch/detekt/blob/91fdc64fecb858045059bf544fa9bfb6017c8c1d/detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt#L123-L135

**Comment by DanySK** [comment]
@BraisGabin @schalkms I realized I was wrongly using version 1.7.1. The behaviour is already fixed in 1.7.2, and this bug is invalid.
Thanks
